ironfish/src/wallet/account.ts

```diff
@@ -157,8 +158,7 @@ export class Account {
 
       await this.accountsDb.saveDecryptedNote(this, noteHash, note, tx)
 
-      const transaction = await this.getTransaction(note.transactionHash, tx)
-      this.saveDecryptedNoteSequence(noteHash, transaction?.sequence ?? null)
+      this.saveDecryptedNoteSequence(noteHash, sequence)
```

Comment:
this does break, beause the wrong sequence is being passed in some places. I haven't tracked it down yet but the test failure is an accurate new bug that's created.